src/heretic/d_main.c

```diff
@@ -162,51 +162,59 @@ void DrawCenterMessage(void)
 //
 //---------------------------------------------------------------------------
 
+int left_widget_w, right_widget_w; // [crispy]
+
 static void CrispyDrawStats (void)
 {
     static short height, coord_x;
     char str[32];
     player_t *const player = &players[consoleplayer];
+    int left_widget_x, right_widget_x;
 
-    if (!height || !coord_x)
+    if (!height || !coord_x || !right_widget_w)
     {
 	const int FontABaseLump = W_GetNumForName(DEH_String("FONTA_S")) + 1;
 	const patch_t *const p = W_CacheLumpNum(FontABaseLump + 'A' - 33, PU_CACHE);
 
 	height = SHORT(p->height) + 1;
-	coord_x = ORIGWIDTH - 7 * SHORT(p->width);
+        right_widget_w = 7 * SHORT(p->width);
+	coord_x = ORIGWIDTH - right_widget_w;
     }
 
+    left_widget_x = 0 - WIDESCREENDELTA;
+    right_widget_x = coord_x + WIDESCREENDELTA;
+
     if (crispy->automapstats == WIDGETS_ALWAYS || (automapactive && crispy->automapstats == WIDGETS_AUTOMAP))
     {
 	M_snprintf(str, sizeof(str), "K %d/%d", player->killcount, totalkills);
-	MN_DrTextA(str, 0, 1*height);
+	MN_DrTextA(str, left_widget_x, 1*height);
+        left_widget_w = MN_TextAWidth(str); // Assume that kills is longest string
```

Comment:
Will this get reset when the widgets are disabled?

Reply:
Whoops, good catch.

Reply:
Well, maybe not that much, because the `left_widget_w` gets only added to `spinfly_x` if the widgets are actually visible.